pkgs/development/libraries/abseil-cpp/202206.nix

```diff
@@ -0,0 +1,36 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, cmake
+, static ? stdenv.hostPlatform.isStatic
+, cxxStandard ? null
+}:
+
+stdenv.mkDerivation rec {
+  pname = "abseil-cpp";
+  version = "20220623.1";
+
+  src = fetchFromGitHub {
+    owner = "abseil";
+    repo = "abseil-cpp";
+    rev = "refs/tags/${version}";
+    hash = "sha256-Od1FZOOWEXVQsnZBwGjDIExi6LdYtomyL0STR44SsG8=";
+  };
+
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=${if static then "OFF" else "ON"}"
+  ] ++ lib.optionals (cxxStandard != null) [
+    "-DCMAKE_CXX_STANDARD=${cxxStandard}"
+  ];
```

Comment:
I understand this was carried over from the previous abseil expression. But, I'm not sure how I feel about this, it should probably be the responsibility of upstream to determine what C++ standard they require.

Reply:
This relates to f9180933cae3dea5546b6954c66da7b77e04ff4c by @fabianhjr

Reply:
And `grpc` fails to build without this override.

Reply:
It's really annoying, because this is completely within the domain of the CODE WHICH IS BEING WRITTEN. -_-

I guess I'm just tired of doing CMake fixes for a lot of "commonly used packages", because release management is usually an afterthought :(

Reply:
Abseil upstream supports being compiled with multiple C++ standards versions; see [absl/base/options.h](https://github.com/abseil/abseil-cpp/blob/lts_2022_06_23/absl/base/options.h). Different upstream projects using Abseil require it to be compiled in different ways. That’s why this needs to be overridable.

Reply:
Feel like we should expose something which is more similar to how it's supposed to be used. Like a stub which does `add_subdirectory(<abseil-cpp.src>)`.

It doesn't really make sense to expose abseil as a "library", if it needs to inherit a bunch of build-time configuraition.

Reply:
or expose "flavors" of abseil for each cxx version. `abseil-cpp.passthru.cxx_17`.

Either way, I still feel like there's a way for `abseilConfig.cmake` to communicate the expected C++ standard